src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/prisonperson/identifyingmarks/IdentifyingMarkImageResource.kt

```diff
@@ -86,6 +95,57 @@ class IdentifyingMarkImageResource(private val service: IdentifyingMarkImageServ
   suspend fun getIdentifyingMarkImageMappings(
     @PathVariable dpsImageId: UUID,
   ) = service.getIdentifyingMarkImageMapping(dpsImageId)
-    ?.toResponse()
+    ?.toDto()
     ?: throw NotFoundException("Identifying mark image mapping not found for DPS image id $dpsImageId")
+
+  @PostMapping("/identifying-mark-image")
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Create an identifying mark image mapping",
+    description = "Create an identifying mark image mapping. Requires ROLE_NOMIS_PRISONPERSON",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = AlertMappingDto::class))],
+    ),
+    responses = [
+      ApiResponse(responseCode = "201", description = "Mapping created"),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Access forbidden for this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Indicates a duplicate mapping has been rejected. If Error code = 1409 the body will return a DuplicateErrorResponse",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = DuplicateMappingErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  suspend fun createIdentifyingMarkImageMapping(
+    @RequestBody mapping: IdentifyingMarkImageMappingDto,
+  ) = try {
+    service.createIdentifyingMarkImageMapping(mapping.toEntity())
+  } catch (e: DuplicateKeyException) {
+    throw DuplicateMappingException(
```

Comment:
seems slightly odd for resource to catch an exception and then throw another one instead, rather than the service to be doing it

Reply:
I thought similar but decided to just follow existing patterns for the sake of consistency. What do you think @andymarke ?